File: leetcode/0219-contains-duplicate-ii/containsNearbyDuplicate.test.ts
import cases from "./cases.json";
import containsNearbyDuplicate from "./containsNearbyDuplicate";

test("contains duplicate ii", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = containsNearbyDuplicate(
      testCase.input.nums,
      testCase.input.k
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toBe(testCase.output);
  });
});

File: leetcode/0219-contains-duplicate-ii/containsNearbyDuplicate_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type ContainsNearbyDuplicateCase struct {
	Input struct {
		Nums []int
		K    int
	}

	Output bool
}

func TestContainsNearbyDuplicateCase(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]ContainsNearbyDuplicateCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := containsNearbyDuplicate(testCase.Input.Nums, testCase.Input.K)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
